src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -662,6 +667,112 @@ private void OnHostLaunched(HostProviderEventArgs e)
+        private string CheckTestHostDllArchitecture(string testHostPath, string muxerPath, PlatformArchitecture targetArchitecture)
+        {
+            // By default compatibility is true, in case of unknown architecture we'll fail.
+            bool isCompatibleTestHostDll = true;
+            using (Stream stream = this.fileHelper.GetStream(testHostPath, FileMode.Open, FileAccess.Read))
+            using (PEReader peReader = new PEReader(stream))
+            {
+                switch (peReader.PEHeaders.CoffHeader.Machine)
+                {
+                    case Machine.IA64:
+                    case Machine.Amd64:
+                        isCompatibleTestHostDll = targetArchitecture == PlatformArchitecture.X64;
+                        break;
+                    case Machine.Arm64:
+                        isCompatibleTestHostDll = targetArchitecture == PlatformArchitecture.ARM64;
+                        break;
+                    case Machine.Arm:
+                        isCompatibleTestHostDll = targetArchitecture == PlatformArchitecture.ARM;
+                        break;
+                    case Machine.I386:
+                        var headers = peReader.PEHeaders;
+                        var characteristics = headers.CoffHeader.Characteristics;
+                        var corflags = headers.CorHeader.Flags;
+
+                        // If requires 32bit or if R2R
+                        if ((corflags & CorFlags.Requires32Bit) != 0 || ((corflags & CorFlags.ILOnly) == 0))
+                        {
+                            isCompatibleTestHostDll = targetArchitecture == PlatformArchitecture.X86;
+                        }
+                        else
+                        {
+                            EqtTrace.Error("DotnetTestHostmanager: AnyCPU testhost.dll provided {0}", testHostPath);
+                            return testHostPath;
+                        }
+
+                        break;
+                    default:
+                        EqtTrace.Error("DotnetTestHostmanager: Unknown architecture {0}", peReader.PEHeaders.CoffHeader.Machine);
+                        break;
+                }
+
+                if (!isCompatibleTestHostDll)
+                {
+                    EqtTrace.Verbose("DotnetTestHostmanager: Incompatible architecture for testhost.dll '{0}', {1}", testHostPath, peReader.PEHeaders.CoffHeader.Machine);
+                }
+            }
+
+            if (!isCompatibleTestHostDll)
+            {
+                string runningSdkVersionPathName = Path.GetFileName(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location));
+
+                if (!SemanticVersioning.TryParse(runningSdkVersionPathName, out SemanticVersioning runningSdkVersion))
+                {
+                    EqtTrace.Verbose($"DotnetTestHostmanager: Invalid sdk location, expected a valid semver version but found '{runningSdkVersionPathName}'");
```

Comment:
Nit: Adding the end of sentence period.

```suggestion
                    EqtTrace.Verbose($"DotnetTestHostmanager: Invalid sdk location, expected a valid semver version but found '{runningSdkVersionPathName}'.");
```